theano/tensor/tests/test_subtensor.py

```diff
@@ -337,6 +339,30 @@ def test_slice_symbol(self):
         ret = f()
         assert ret.shape == (1, 1, 4)
 
+    def test_ellipsis(self):
+        numpy_n = numpy.arange(24, dtype=self.dtype).reshape((2, 3, 4))
+        n = self.shared(numpy_n)
+        for length, op_type, slice_ in [
+                (0, self.sub, numpy.index_exp[...]),
+                (1, self.sub, numpy.index_exp[..., 1]),
+                (1, self.sub, numpy.index_exp[1, ...]),
+                (1, self.sub, numpy.index_exp[..., 1, 2, 3]),
+                (1, self.sub, numpy.index_exp[1, ..., 2, 3]),
+                (1, self.sub, numpy.index_exp[1, 2, 3, ...]),
+                (3, DimShuffle, numpy.index_exp[..., [0, 2, 3]]),
+                (1, DimShuffle,
+                 numpy.index_exp[numpy.newaxis, ...]),
+                (1, AdvancedSubtensor,
+                 numpy.index_exp[..., numpy.newaxis, [1, 2]])]:
```

Comment:
Older versions of numpy do not support `numpy.newaxis` with advanced indexing, which is why that test was failing with numpy 1.7.
An option may be to skip this test for numpy < 1.9 rather than bumping the requirements.

If we decide to change the requirements, then we should also update the doc pages and requirements.txt files that mention the older version.